expo/expo
# [dev-launcher][iOS] update sourceUrl type to be nullable

# Why



We (Eric and I) were able to reliably reproduce a crash after selecting "Reload" on the dev launcher error screen

This occurs under fairly specific circumstances - the app is freshly installed (no prompt for local network permissions yet) and is handling a deep link from a cold boot (e.g dev client is not running in the background)


# How



The important part of the ErrorViewController reload method looks like this: 

```
   guard let appUrl = manager?.controller?.sourceUrl() else {
      // We don't have app url. So we fallback to launcher.
      // Shoudn't happen.
      navigateToLauncher()
      return
    }
```

Where accessing the sourceUrl property was leading to this error: 
```
EXC_BREAKPOINT (code=1, subcode=0x19255ccc8)
```

After some trial and error, adding a nullable to this getter method seemed to help with the interop between swift and objc and fix the issue

# Test Plan



- Install an app w/ dev-client SDK 45 onto a physical iOS device
- Start an app w/ `expo start` 
- Ensure that the dev launcher app is closed (not backgrounded)
- Launch the app via QR code 
- Press `Reload` and ensure that the app navigates to the dev launcher Home Screen, rather than crashing


# Checklist
